### ISSUE TITLE
Installs oauth2client 1.4.12, then complains it's too old.

### ISSUE THREAD

**Issue by markscudder** [insanum/gcalcli#258]
New stock Debian Jessie in a VM, attempting to prototype before moving to Raspbian. Using your script because it's apparently the only script on earth that will print a Google Calendar event list for Conky. I don't know Python. Google doesn't publish an API wrapper for Perl.

When I run your setup.py as instructed on your WordPress site (yes I followed the instructions for Debian), it "processes dependencies" for gcalcli, and then goes looking for oauth2client 1.4.12. It finds and installs oauth2client 1.4.12, and then immediately dies, complaining that your script needs oauth2client 1.5.0 or higher. Using system Python, which is version 2.7.9.

This sort of thing makes me hate Linux and FOSS in general, especially since not everybody has the kind of mind to do programming, or learn a new language every time somebody's API isn't supported by the languages you kind of know. Output of setup.py follows:

`
$ sudo python setup.py install
running install
running bdist_egg
running egg_info
writing requirements to gcalcli.egg-info/requires.txt
writing gcalcli.egg-info/PKG-INFO
writing top-level names to gcalcli.egg-info/top_level.txt
writing dependency_links to gcalcli.egg-info/dependency_links.txt
reading manifest file 'gcalcli.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
writing manifest file 'gcalcli.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
warning: install_lib: 'build/lib.linux-x86_64-2.7' does not exist -- no Python modules to install

creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-2.7/gcalcli -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/gcalcli to 755
copying gcalcli.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gcalcli.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gcalcli.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gcalcli.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gcalcli.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
creating 'dist/gcalcli-3.4.0-py2.7.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing gcalcli-3.4.0-py2.7.egg
Removing /usr/local/lib/python2.7/dist-packages/gcalcli-3.4.0-py2.7.egg
Copying gcalcli-3.4.0-py2.7.egg to /usr/local/lib/python2.7/dist-packages
gcalcli 3.4.0 is already the active version in easy-install.pth
Installing gcalcli script to /usr/local/bin

Installed /usr/local/lib/python2.7/dist-packages/gcalcli-3.4.0-py2.7.egg
Processing dependencies for gcalcli==3.4.0
Searching for oauth2client<=1.4.12
Reading https://pypi.python.org/simple/oauth2client/
Best match: oauth2client 1.4.12
Downloading https://pypi.python.org/packages/4a/de/9cc7e74bc217b9f428945b3c85ad83f5ab21ec77718408cfe8ab36118938/oauth2client-1.4.12-py2.7.egg#md5=8265f6985905a5590c622741980a47e5
Processing oauth2client-1.4.12-py2.7.egg
Moving oauth2client-1.4.12-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Adding oauth2client 1.4.12 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/oauth2client-1.4.12-py2.7.egg
error: Installed distribution oauth2client 1.4.12 conflicts with requirement oauth2client>=1.5.0,<4.0.0
`

**Comment by tresni** [comment]
Recommended install is just through pip, but it looks like you have some other package that needs a newer version of oauthclient the we can support in the 3.x branch.

If you are okay with early release do `pip install gcalcli==4.0.0a1` which works with later releases of oauthclient.

-- Brian

**Comment by markscudder** [comment]
Installing 4.0.0a1 gets me past the oauth2client issue, but only outputs "ERROR: Missing module - No module named discovery."

Since I don't know Python (and feel strongly that I shouldn't have to in order to install working software, in fact if I could make this work it might motivate me to learn some Python), all I can pretend to know to do is "pip install discovery." Pip finds and installs a package called "discovery," but gcalcli is apparently looking for a _different_ package called "discovery," as the error persists.

Your observation that another package on my system needs oauth2client 1.4.12 is incorrect - this is a fresh, clean Debian Jessie 64-bit install on a VM (specifically VMWare Fusion on a Mac, but that shouldn't matter). The only things I've installed are Conky, Conky Manager, and some stuff from CPAN. After failing to build gcalcli the first time I attempted to install the Google API from pip, but was told it was already installed (probably because of gcalcli). But that's it. The install is literally two days old. So I can't imagine, unless something that comes stock with Jessie requires 1.4.12, that I did anything to bork my Python install (and if I did, shame on Python, because I don't even know enough Python to be dangerous, and if it's that fragile, God help us all).

You also said "Recommended install is just through pip." I found this app through a Google search, which took me to the author's blog, where he instructs to apt-get install (for Debian) a bunch of prerequisites, then clone gcalcli from git and run sudo setup.py install, so that's what I did. At this point running "sudo pip install gcalcli" produces this:

`
$ sudo pip install gcalcli
Traceback (most recent call last):
  File "/usr/bin/pip", line 9, in <module>
    load_entry_point('pip==1.5.6', 'console_scripts', 'pip')()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 356, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2476, in load_entry_point
    return ep.load()
  File "/usr/lib/python2.7/dist-packages/pkg_resources.py", line 2190, in load
    ['__name__'])
  File "/usr/lib/python2.7/dist-packages/pip/__init__.py", line 74, in <module>
    from pip.vcs import git, mercurial, subversion, bazaar  # noqa
  File "/usr/lib/python2.7/dist-packages/pip/vcs/mercurial.py", line 9, in <module>
    from pip.download import path_to_url
  File "/usr/lib/python2.7/dist-packages/pip/download.py", line 25, in <module>
    from requests.compat import IncompleteRead
ImportError: cannot import name IncompleteRead
`

So I don't know _what_ I've done now.

I have some family stuff to attend to, but after that I'm going to put a fresh install of Raspbian on my Pi3, which was the target system in the first place (but Raspbian = Debian and you can't run ARM code in an Intel-based VM, etc.) and simply try to install it (first the current version, then the alpha) from Pip.

In the meantime, could you help me understand:

1. How much of this should I be doing as root and/or through sudo? I tend to use sudo when installing anything, including libraries and software. Just an old habit based on where I usually think system software and applications ought to be installed but I wonder if it's causing problems here.
2. Since I do want to learn, if we pretend for a moment I can't just blast out this VM and start over, what would I do to clean up all this crap I've tried ("pip uninstall"?) and get back to a clean Python environment so I could pip install gcalcli as recommended, without all this other stuff stabbing me in the back?

Thanks for your reply.
-m

**Comment by markscudder** [comment]
I have an update.

My pip stopped working (something about a traceback), so following instructions on a StackOverflow post, I did:

```
sudo apt-get remove python-pip
```

Then:

```
sudo easy_install pip
```

This got me a working Pip again, so I did:

```
sudo pip uninstall gcalcli
```

And:

```
sudo pip uninstall google-api-python-client
```

Then:

```
sudo pip install gcalcli
```

And I now have a working gcalcli 3.4.0 on my Jessie VM, at least.

I guess the instructions for building from GitHub should be amended so others don't run in to this. Now I have to follow the instructions for authenticating with Google, but I might be good to go. I'll let you know.

-m
